src/index.ts

```diff
@@ -84,6 +101,28 @@ const kDefaultOptions : FilledOptions = {
   useAtomics: true
 };
 
+class DirectlyTransferable implements Transferable {
+  #value : object;
+  constructor (value : object) {
+    this.#value = value;
+  }
+
+  get [kTransferable] () : object { return this.#value; }
+
+  get [kValue] () : object { return this.#value; }
+}
+
+class ArrayBufferViewTransferable implements Transferable {
+  #view : ArrayBufferView;
+  constructor (view : ArrayBufferView) {
+    this.#view = view;
```

Comment:
Should we verify that `view.buffer.byteLength === view.byteLength` here, to make this less easy to get wrong?

(Aside, since these classes are internal and not exposed to users, I don’t think you really need to make the properties private – currently, that still transpiles to a WeakMap implementation)

Reply:
I don't think so because there are cases (albeit edge cases for sure) where that check doesn't work. I'd say let's leave this and maybe reference the warning we just added to the core docs.

And yeah, these properties don't need to be private

Reply:
Hm, okay… I know there are edge cases, but those seem kinda fabricated to me ;)